luigi/contrib/k8s_job.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2015 Outlier Bio, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+"""
+Kubernetes Job wrapper for Luigi.
+
+From the Kubernetes website:
+
+    Kubernetes is an open-source system for automating deployment, scaling,
+    and management of containerized applications.
+
+For more information about Kubernetes Jobs: http://kubernetes.io/docs/user-guide/jobs/
+
+Requires:
+
+- pykube: ``pip install pykube``
+
+Written and maintained by Marco Capuccini (@mcapuccini).
+"""
+
+import luigi
+from luigi import configuration
+import logging
+import uuid
+import time
+
+try:
+    from pykube.config import KubeConfig
+    from pykube.http import HTTPClient
+    from pykube.objects import Job
+except ImportError:
+    logger.warning('pykube is not installed. KubernetesJobTask requires pykube.')
```

Comment:
logger isn't defined at this point, therefore this attempted warning will cause an exception. Might want to set
`logger = logging.getLogger('luigi-interface')` right after `import time`